packages/core/src/base-chart.ts

```diff
@@ -554,9 +579,30 @@ export class BaseChart {
 					btnsWidth += btn.clientWidth;
 				}
 			});
+		}
+
+		let visibleLegendItems = 0;
+		let totalLegendItems = 0;
+
+		// Count visible legend items
+		this.container.selectAll(".legend-btn.clickable").each(function(d) {
+
+			totalLegendItems++;
+
+			const node = select(this);
+
+			if (node.style("display") === "inline-block") {
```

Comment:
Is there a better way for testing this?
If the intention is "not visually hidden" maybe "not none" would allow us to use other display values in future without breaking this piece.